protocols/v2/noise-sv2/src/error.rs

```diff
@@ -1,18 +1,47 @@
+//! # Noise Sv2 Error Handling
+//!
+//! [`Error`] defines the set of possible errors that may arise during the Noise protocol
+//! handshake, cipher selection, and encryption/decryption processes.
```

Comment:
this is a private module

`Error` is re-exported somewhere else, so those comments can remain `///`

but the header should be `//`